src/contract.py

```diff
@@ -39,6 +39,9 @@ def to_dict(self) -> dict[str, any]:
             'observations': self.num_mon
         }
 
+    def set_num_mon(self, num_mon) -> list[float]:       # used by BB Barrier pricer
```

Comment:
I would simply go with setting this property wherever you need. No need for this function.